main.tf

```
@@ -111,7 +111,7 @@ data "aws_iam_policy_document" "assume_role" {
 
     principals {
       type        = "Service"
-      identifiers = ["rds.${data.aws_partition.current.dns_suffix}"]
+      identifiers = ["rds.amazonaws.com"]
```

lets do this instead

```suggestion
      identifiers = distinct(["rds.${data.aws_partition.current.dns_suffix}", "rds.amazonaws.com"])
```

---

Hi I applied code you suggested and I am getting the following error
```
{"@level":"error","@message":"Error: creating IAM Role (bqo8b2ey-proxy): operation error IAM: CreateRole, https response error StatusCode: 400, RequestID: a57082c6-9ff5-45c3-9ef3-35dd8afcade7, MalformedPolicyDocument: Invalid principal in policy: \"SERVICE\":\"rds.amazonaws.com.cn\"","@module":"terraform.ui","@timestamp":"2025-05-15T16:54:21.610463Z","diagnostic":{"severity":"error","summary":"creating IAM Role (bqo8b2ey-proxy): operation error IAM: CreateRole, https response error StatusCode: 400, RequestID: a57082c6-9ff5-45c3-9ef3-35dd8afcade7, MalformedPolicyDocument: Invalid principal in policy: \"SERVICE\":\"rds.amazonaws.com.cn\"","detail":"","address":"module.minimal.module.rds_proxy.aws_iam_role.this[0]","range":{"filename":".terraform/modules/minimal.rds_proxy/main.tf","start":{"line":119,"column":32,"byte":3997},"end":{"line":119,"column":33,"byte":3998}},"snippet":{"context":"resource \"aws_iam_role\" \"this\"","code":"resource \"aws_iam_role\" \"this\" {","start_line":119,"highlight_start_offset":31,"highlight_end_offset":32,"values":[]}},"type":"diagnostic"}
```
so its the same as before it sends incorrect principal. in the bellow commend i suggested 100% reliable approach. please take a look.